indexing-service/src/main/java/io/druid/indexing/common/TaskLock.java

```diff
@@ -31,19 +31,25 @@
   private final String dataSource;
   private final Interval interval;
   private final String version;
+  private final Integer priority;
+  private final Boolean preemptive;
 
   @JsonCreator
   public TaskLock(
       @JsonProperty("groupId") String groupId,
       @JsonProperty("dataSource") String dataSource,
       @JsonProperty("interval") Interval interval,
-      @JsonProperty("version") String version
+      @JsonProperty("version") String version,
+      @JsonProperty("priority") Integer priority,
+      @JsonProperty("preemptive") Boolean preemptive
```

Comment:
@pjain1 I looked at the older conversation and it turns out I'm the one that didn't like "exclusive" :P

I had suggested "non-preemptible" or "uninterruptible" but now after reading everything, I think the best word is actually "upgraded". That should make most of the if checks easier to read, since it should avoid most of the double-negatives…